### ISSUE TITLE
When presses replay button karma points addition isn't correct and adds power karma points

### ISSUE THREAD

**Issue by aditmehta9** [systers/powerup-android#1394]
## Description
As a user, I need to reset karma points count properly so that I can use karma points correctly.

## Mocks
![ezgif com-video-to-gif](https://user-images.githubusercontent.com/35133994/76698215-33647200-66c6-11ea-8d48-e999079c0474.gif)

### Update [Required]
- [ ] Update karma points to (whole new + previous points) when the user plays reset button, while in this case only power points are only added after the replay. Besides dialog box doesn't show the addition of karma points of powers.

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 1 mentor.

## Estimation
4 hours

**Comment by aditmehta9** [comment]
@sunjunkie  @Rimjhim28  @isabelcosta @mayburgos  Can I work on this issue?

**Comment by sammy1997** [comment]
Approving the issue, but as you are already assigned to 2 issues, I am making this available for others.

**Comment by devanshi7799** [comment]
@sammy i would like to work on this one

**Comment by vold-la** [comment]
@sammy1997  Can i work on this?

**Comment by HaripriyaB** [comment]
Can I work on this pls? I haven't got a chance yet. @sammy1997 @mayburgos @sunjunkie

**Comment by robotjellyzone** [comment]
Hi @HaripriyaB , if you are a first time contributor here then please take a look at the issues with labels **First Timer Only**  and only those not assigned yet .

**Comment by HaripriyaB** [comment]
@robotjellyzone yes I'm a first timer and wish to work on this project. I'll look in to the first timer tag. Thanks!

**Comment by sammy1997** [comment]
@vold-la go ahead

**Comment by vold-la** [comment]
I think this not not an issue but a mere confusion to understand the game. First the dialog box at the end does not show addition of power points since power points earned are directly added to total points which acts as bonus in the game.
Secondly if we press replay button then karma points(only earned by giving answers) earned from previous play are deducted. Let us take example, you are having 20 points and you played a scenario and earned 6 points so your  total becomes 26. Now you replay the scenario so 6 points get deducted so your total becomes 20, then you earned 20  power points and 6 points by giving response. so again total becomes 46.
So to avoid this confusion we can prompt dialog informing "if you replay your previous point will get deducted" and add in current dialog of power points that power points will be added to total points. 
what do you think? is this correct? should i open other issue to add dialog? @sammy1997  @aditmehta9

**Comment by aditmehta9** [comment]
> 
> 
> I think this not not an issue but a mere confusion to understand the game. First the dialog box at the end does not show addition of power points since power points earned are directly added to total points which acts as bonus in the game.
> Secondly if we press replay button then karma points(only earned by giving answers) earned from previous play are deducted. Let us take example, you are having 20 points and you played a scenario and earned 6 points so your total becomes 26. Now you replay the scenario so 6 points get deducted so your total becomes 20, then you earned 20 power points and 6 points by giving response. so again total becomes 46.
> So to avoid this confusion we can prompt dialog informing "if you replay your previous point will get deducted" and add in current dialog of power points that power points will be added to total points.
> what do you think? is this correct? should i open other issue to add dialog? @sammy1997 @aditmehta9

Yeah, that is right but I also wanted to show the addition of points gained by powers to show in a dialog box. But according to your logic if a user plays level 1 scenario 3 hospital and after playing that scenario his total points are 34 and in the next scenario, he clicks on the home button without playing scenario 4(library) and replays scenario 1(home) then new points are added without deduction of previous points and the total becomes 44. However, which should not happen according to your logic and deduct points of playing that scenario when it was played earlier.

**Comment by aditmehta9** [comment]
So, I think there should be a change in the game logic for all scenarios replay or last played scenario.

**Comment by vold-la** [comment]
totally agree that points section needs to be organized. We should discuss this with mentors in zulip.

**Comment by aditmehta9** [comment]
> 
> 
> totally agree that points section needs to be organized. We should discuss this with mentors in zulip.

Yep

**Comment by aditmehta9** [comment]
> 
> 
> > I think this not not an issue but a mere confusion to understand the game. First the dialog box at the end does not show addition of power points since power points earned are directly added to total points which acts as bonus in the game.
> > Secondly if we press replay button then karma points(only earned by giving answers) earned from previous play are deducted. Let us take example, you are having 20 points and you played a scenario and earned 6 points so your total becomes 26. Now you replay the scenario so 6 points get deducted so your total becomes 20, then you earned 20 power points and 6 points by giving response. so again total becomes 46.
> > So to avoid this confusion we can prompt dialog informing "if you replay your previous point will get deducted" and add in current dialog of power points that power points will be added to total points.
> > what do you think? is this correct? should i open other issue to add dialog? @sammy1997 @aditmehta9
> 
> Yeah, that is right but I also wanted to show the addition of points gained by powers to show in a dialog box. But according to your logic if a user plays level 1 scenario 3 hospital and after playing that scenario his total points are 34 and in the next scenario, he clicks on the home button then new points are added without deduction of previous points and total becomes 44. However, which should not happen according to your logic and deduct points of playing that scenario when it was played earlier.
